### ISSUE TITLE
TypeError: undefined is not an object (evaluating 'this.store.firestore')

### ISSUE THREAD

**Issue by pennjammin** [prescottprue/react-redux-firebase#594]
**Do you want to request a *feature* or report a *bug*?**  
Bug

**What is the current behavior?**
When following the migration guide (I changed everything in the diff) and ensuring that I upgraded my react-redux-firbase package to v3.0.0-alpha, those changes lead to the following error. Both my Firebase and Firestore objects are empty when I var dump the state.

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem via [codesandbox](https://codesandbox.io/) or similar.**

**App.js**
```javascript
import React from 'react';
import { createStore, combineReducers, compose, applyMiddleware } from 'redux'
import { Provider } from 'react-redux'
import thunk from 'redux-thunk'

import { Platform, StatusBar, StyleSheet, View } from 'react-native'
import { Asset, Font, Icon } from 'expo'
import AppNavigator from './navigation/AppNavigator'
import rootReducer from './reducers'

import firebase from 'firebase/app'
import 'firebase/auth'
import 'firebase/database'
import 'firebase/firestore' // make sure you add this for firestore
import { ReactReduxFirebaseProvider, getFirebase } from 'react-redux-firebase'
import { createFirestoreInstance, getFirestore } from 'redux-firestore'

//firebase configuration
const firebaseConfig = {
  ...
};

// react-redux-firebase options
const rrfConfig = {
  userProfile: 'users', // firebase root where user profiles are stored
  useFirestoreForProfile: true, // Firestore for Profile instead of Realtime DB
  enableLogging: false, // enable/disable Firebase's database logging
}

// Initialize Firebase
firebase.initializeApp(firebaseConfig);

const createStoreWithFirebase = compose(
  applyMiddleware(thunk.withExtraArgument({ getFirebase, getFirestore }))
)(createStore);

// Create store with reducers and initial state
const initialState = {}
const store = createStoreWithFirebase(rootReducer, initialState);

const rrfProps = {
  firebase,
  config: rrfConfig,
  dispatch: store.dispatch,
  createFirestoreInstance // <- needed if using firestore
}

export default class App extends React.Component {

  static navigationOptions = {
    header: null
  }

  render() {
      return (
        <Provider store={store}>
          <ReactReduxFirebaseProvider {...rrfProps}>
            <View style={styles.container}>
              {Platform.OS === 'ios' && <StatusBar barStyle="default" />}
              <AppNavigator />
            </View>
          </ReactReduxFirebaseProvider>
        </Provider>
      )
  }

  _loadResourcesAsync = async () => {
    return Promise.all([
      Asset.loadAsync([
        require('./assets/images/logo.png'),
      ]),
      Font.loadAsync({
        // This is the font that we are using for our tab bar
        ...Icon.Ionicons.font,
        // We include SpaceMono because we use it in HomeScreen.js. Feel free
        // to remove this if you are not using it in your app
        'space-mono': require('./assets/fonts/SpaceMono-Regular.ttf'),
      }),
    ])
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
  },
})

```

**SomeComponent.js**

```javascript
const mapStateToProps = (state) => {
  console.log(state);
  return {
    user: state.firestore.users
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    addPoint: () => dispatch(addPoint())
  }
}

export default compose(
  firestoreConnect([
    { collection: 'users' }
  ]),
  connect(mapStateToProps, mapDispatchToProps)
)(SomeComponent)
```

**What is the expected behavior?**
Before I had state logging the Firebase object w/ it full of data. All I want is to hook into the Firestore and get data with the new library. I should be able to keep my actions the same I assume. It seems I am having an issue with the compose/connect but can not for the life of me understand why. My apologies, this migration just has me in a tizzy, haha. Could it have to do with the thunk middleware? Or am I just missing something. Am I not supposed to use the firestoreConnect as it was the breaking feature before?? And instead us withFirestore? Thanks so much for any insight. I will continue to troubleshoot.

**Which versions of dependencies, and which browser and OS are affected by this issue? Did this work in previous versions or setups?**
macOS mojave, react-native in expo

**Comment by prescottprue** [comment]
@pennjammin What are your dependency versions? Did you make sure to upgrade `react-redux` to 6 and `react` to `^16.4`? You may not be able to keep your actions the same if you were using `getFirebase` or `getFirestore`. If you post your action creator code, but it doesn't seem like that is it.

Something to note is that `getFirebase` and `getFirestore` are no longer part of the API in v3 (this may change, but since context works differently, those utilities had to be removed until they can be updated).


Side Note unrelated to the original question:

`mapDispatchToProps` does just that. It maps your action creators with dispatch, so you could just do:

```js
const mapDispatchToProps = (dispatch) => {
  return {
    addPoint
  }
}
```

Then that action creator will be mapped with dispatch.

**Comment by pennjammin** [comment]
@prescottprue Thanks so much for responding! Such a well maintained package, haha. So I did check to make sure I was indeed running v6.0.0. and `react` to `^16.4`. So after you mentioned this I think that it does indeed have something to do with passing getFirebase and getFirestore to the thunk middleware. Any suggestions on what to pass in instead? Also, cheers on the tip, I will be sure to change that!!

**Comment by pennjammin** [comment]
@prescottprue Cool! Yeah I just had an arbitrary UID in there for testing, should have been clearer, my b. I still get the error but I will persist and report back. May be silly, but if I use the above method then would I just use mapDispatchToProps for whatever actions aren't handlers?? I am going to go pour back over redux and your docs haha. Still a bit new to this all and despite having it working a few days ago the v3 thing through me for a loop. I have added the error for posterity as well:
![img_f4ad9130415d-1](https://user-images.githubusercontent.com/40776807/50120658-43bc4180-0224-11e9-878b-d5d6fa30b322.jpeg)

**Comment by prescottprue** [comment]
Upon further investigation it seems like there was still some code using the old context pattern in `firestoreConnect`, so I released a fix in [v3.0.0-alpha.3](https://github.com/prescottprue/react-redux-firebase/releases/tag/v3.0.0-alpha.3). Going to try to get unit tests back in place for the v3 features to catch stuff like this sooner.

Let me know if things don't work as expected now and we can debug further. I'm also often available [on gitter](https://gitter.im/redux-firebase/Lobby) if you have implementation questions. Thanks again for reporting.

**Comment by AOUNT** [comment]
Hello @prescottprue ,

Sorry to ask but I'm facing an issue when migrating. I'm receiving this error:
![image](https://user-images.githubusercontent.com/27626560/63642107-5d1e6300-c6ba-11e9-9f6a-cba6973bea90.png)

I ensured to use the correct versions:
      "react": "^16.9.0",
     "react-redux": "^7.1.0",
     "react-redux-firebase": "^3.0.0-alpha.16",

Before using firestoreConnect, when logging state I see this:

<img width="663" alt="Capture d’écran 2019-08-24 à 22 01 17" src="https://user-images.githubusercontent.com/27626560/63642120-c43c1780-c6ba-11e9-8373-063b2c2d8ac3.png">

I'm using a simple component:

`const mapStateTopProps = state => {
    console.log(state);
    return {
        orders :  state.firestore.data.orders
    };
}`

`export default compose(
    firestoreConnect([
        { collection: 'orders' }
      ]), 
    connect(mapStateTopProps)
  )(FoodBuilder) `

I got rid of old getFirestore and getFirebase. 

my index.js looks like this:

![Capture d’écran 2019-08-24 à 22 07 59](https://user-images.githubusercontent.com/27626560/63642170-bf2b9800-c6bb-11e9-8d02-c731af9322f9.png)

it's been 2 days i'm trying to debug :(

thank you in advance !
